apps/deploy_monitoring/src/main.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) SimpleStaking and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+#![forbid(unsafe_code)]
+
+use std::collections::VecDeque;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::{Arc, RwLock};
+
+use slog::{info, Drain, Level, Logger};
+use tokio::signal;
+
+mod configuration;
+mod deploy_with_compose;
+mod display_info;
+mod image;
+mod monitors;
+mod node;
+mod rpc;
+mod slack;
+
+use crate::monitors::resource::{ResourceUtilization, MEASUREMENTS_MAX_CAPACITY};
+use crate::monitors::{
+    shutdown_and_cleanup, start_deploy_monitoring, start_resource_monitoring, start_sandbox,
+    start_sandbox_monitoring, start_stack,
+};
+
+#[tokio::main]
+async fn main() {
+    // parse and validate program arguments
+    let env = configuration::DeployMonitoringEnvironment::from_args();
+
+    // create an slog logger
+    let log = create_logger(env.log_level);
+
+    let slack_server = slack::SlackServer::new(
+        env.slack_url,
+        env.slack_token,
+        env.slack_channel_name,
+        log.clone(),
+    );
+
+    let running = Arc::new(AtomicBool::new(true));
+    let mut thread_handles = Vec::new();
```

Comment:
@adonagy 
I would maybe refactored it to separate fuctionsm which returns thread_handles, like:
```
let thread_handles = if env.is_sandbox {
    launch_sandbox(...)
} else {
    launch_node_stack(...)
};
```